chisel/app.py

```
@@ -208,6 +208,33 @@ def __init__(self, app, environ=None, start_response=None, url_args=None):
             handler.setFormatter(logging.Formatter(self.app.log_format))
         self.log.addHandler(handler)
 
+    @property
+    def reconstructed_url(self):
+        """
+        Reconstructs the request URL using the algorithm provided by PEP3333
+        """
+        environ = self.environ
+        try:
+            url = environ['wsgi.url_scheme']+'://'
+            if environ.get('HTTP_HOST'):
+                url += environ['HTTP_HOST']
+            else:
+                url += environ['SERVER_NAME']
+                if environ['wsgi.url_scheme'] == 'https':
+                    if environ['SERVER_PORT'] != '443':
+                       url += ':' + environ['SERVER_PORT']
+                else:
+                    if environ['SERVER_PORT'] != '80':
+                       url += ':' + environ['SERVER_PORT']
+            url += quote(environ.get('SCRIPT_NAME', ''))
+            url += quote(environ.get('PATH_INFO', ''))
+            if environ.get('QUERY_STRING'):
+                url += '?' + environ['QUERY_STRING']
+            return url
+        except Exception as e:
```

what is this guarding against?

---

This code is directly copied from the PEP

---

doesn't mean its not poopy...

---

I'm swallowing the exception because if any of the necessary bits of environ aren't present it means the context isn't inside a well-formed HTTP request and therefore the URL is logically None.

---

Feel free to remove if you don't like it.

---

ah I see the PEP doesn't have the exception handler - I'll check their code and make sure its sane - and get it up on pypi in a bit.